src/clouds/owncloud.js

```diff
@@ -0,0 +1,129 @@
+/*******************************************************************************
+ * Copyright (C) 2018 Orange.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License 2.0
+ * which accompanies this distribution, and is available at
+ * https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *    Rombit - initial API and implementation
+ ******************************************************************************/
+
+// const debug = require('debug-levels')('agile-data');
+// const agile = require('./agile-sdk');
+// const get = require('lodash/get');
+const path = require('path');
+const request = require('request');
+
+module.exports = {
+  endpointDescription: function () {
+    return({
+      "cloud": "owncloud",
+    })
+  },
+  upload: function(credentials, remotepath,payload) {
+    return new Promise(function(resolve, reject) {
+
+      if (typeof remotepath !== 'undefined' && typeof remotepath !== "undefined"  && typeof payload !== "undefined") {
+      let uploadConfig = {
+        uploadContent: JSON.stringify(payload),
+        uploadFileName: (typeof remotepath !== 'undefined') ? remotepath : dateFormat(new Date(), "yyyy.mm.dd") + "-Agile.txt",
+        uploadDirectory: (typeof remotepath !== 'undefined') ? path.dirname(remotepath) : '',
+        uploadDirectories: (typeof remotepath !== 'undefined') ? path.dirname(remotepath).split('/') : [],
+        loopCount: 0
+      };
+        console.log("--- Owncloud upload data ---");
+      console.log(uploadConfig);
+
+      uploadToOwnCloud(uploadConfig, credentials, function (err) {
+        if (err) {
+          console.log("Owncloud: error uploading file");
+          reject(err);
+        }else{
+          resolve( "upload complete")
+        }
+      });
+
+      } else {
+        reject("Owncloud: request invalid.");
+      }
+    });
+  }
+};
+function uploadToOwnCloud(uploadConfig, credentials, cb) {
+  let req = {
+    method: 'PUT',
+    uri: credentials.owncloudServer + '/remote.php/webdav/'+ uploadConfig.uploadFileName,
+    auth: {
+      username: credentials.clientId,
+      password: credentials.clientSecret
+    },
+    timeout: 10000,
+    body: uploadConfig.uploadContent
+  };
+
+  request(req, function (error, response, body) {
+    if (error && (typeof response === "undefined" || typeof response.statusCode === "undefined")) {
```

Comment:
Can be replaced with `if (error && (!response || !response.statusCode))`

This can throw an error though, if `response` is `undefined` then when `typeof response.statusCode === 'undefined' ` is evaluated, an error will be thrown, because the value `statusCode` of `undefined` can not be read

Reply:
Fixed as well, the code looks slightly different now because of axios.